Assets/Scripts/Combat/PopupTextController.cs

```diff
@@ -0,0 +1,26 @@
+﻿using UnityEditor;
+using UnityEngine;
+
+public class PopupTextController : MonoBehaviour
+{
+    private static string popupTextPrefabFilePath = "Assets/Prefabs/PopupTextParent.prefab";
+    private static PopupText popupTextPrefab;
+    private static string canvasName = "Canvas";
+    private static GameObject canvas;
+
+    public static void Initialize()
+    {
+        canvas = GameObject.Find(canvasName);
+        popupTextPrefab = AssetDatabase.LoadAssetAtPath<PopupText>(popupTextPrefabFilePath);
+    }
+
+    public static void CreatePopupText(string text, Transform transform)
+    {
+        PopupText instance = Instantiate(popupTextPrefab);
+        Vector2 screenPosition = Camera.main.WorldToScreenPoint(transform.position);
```

Comment:
Some games have the popup text appear at an elevated Y coordinate relative to the unit - up to you if you think it looks better that way or not.

Reply:
I considered that, but all the units have status bars above their heads so I wasn't sure where to put the text. It'll be easy to change later once there's a firmer direction for the UI.